lib/plugins/aws/package/compile/events/activemq.js

```diff
@@ -106,16 +109,17 @@ class AwsCompileActiveMQEvents {
           functionName,
           queue
         );
-        const lambdaLogicalId = this.provider.naming.getLambdaLogicalId(functionName);
-        const dependsOn = this.provider.resolveFunctionIamRoleResourceName(functionObj) || [];
+
+        const dependsOn = _.compact([
+          this.provider.resolveFunctionIamRoleResourceName(functionObj),
+          _.get(functionObj.targetAlias, 'logicalId'),
```

Comment:
Why we add here function to depends on? This feels not needed and out of scope of that issue

Reply:
Adding the function alias to DependsOn is needed (particularly for snapStart as they take a long time to create - not sure about provisionedCapacity) because CloudFormation will fail on deploy without it saying it can't find the alias.  In my current serverless project I had to add these DependsOn in resources: extensions: to get it to work.

If you look at sqs.js or stream.js (which I didn't change in this PR) you'll see this DependsOn is there in those events.